platforms/core-execution/build-cache/src/main/java/org/gradle/caching/local/internal/DirectoryBuildCacheServiceFactory.java

```diff
@@ -95,8 +92,8 @@ public BuildCacheService createBuildCacheService(DirectoryBuildCache configurati
             .cache(target)
             .withCleanupStrategy(createCacheCleanupStrategy(removeUnusedEntriesOlderThan))
             .withDisplayName("Build cache")
-            .withLockOptions(mode(OnDemand))
-            .withCrossVersionCache(CacheBuilder.LockTarget.DefaultTarget)
+            .withLockOptions(new DefaultLockOptions())
+            .withCrossVersionCache()
```

Comment:
Does `withCrossVersionCache` really affect the same thing as `withLockOptions`?  `wCVC` sets the `versionStrategy` on the cache-to-be-built.  This isn't something configured by `LockOptions`.

`withCrossVersionCache is sometimes used without the cross-version flag being set in lock options.  This predates any of my changes, see `CrossBuildCachingKeyService` or `ExecutionGradleServices` for instance.  This was and is fine because by default the lock files are created in the (cross version) content dir, but it's why I can't add validation that you set the flag when you call the method - it looks like it is intentional that these two things don't have to remain in sync.

The default lock option values are different in `DefaultLockOptions` and in the default lock options instance used by `PersistentCacheBuilder` - this is preserving existing behavior but is weird.

I'm now much less sure about pushing lock options down into the `CacheBuilder` itself after looking more closely at all this.